app/src/main/java/com/github/sdp/ratemyepfl/viewmodel/ClassroomsListViewModel.kt

```diff
@@ -1,29 +1,16 @@
 package com.github.sdp.ratemyepfl.viewmodel
 
 import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.ViewModel
-import androidx.lifecycle.viewModelScope
-import com.github.sdp.ratemyepfl.database.ItemsRepositoryInterface
+import com.github.sdp.ratemyepfl.database.ItemRepository
 import com.github.sdp.ratemyepfl.model.items.Classroom
 import dagger.hilt.android.lifecycle.HiltViewModel
-import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
-class ClassroomsListViewModel @Inject constructor(private val repository: ItemsRepositoryInterface) :
-    ViewModel() {
-
-    private var roomsLiveData = MutableLiveData<List<Classroom>>()
-
-    init {
-        viewModelScope.launch {
-            roomsLiveData.value = repository.getClassrooms()
-        }
-    }
+class ClassroomsListViewModel @Inject constructor(repository: ItemRepository<Classroom>) : ReviewableListViewModel<Classroom>(repository){
```

Comment:
What is this class used for ? Couldn't we use directly ReviewableListViewModel<Classroom> ?

Reply:
I had it to not change too many files, but I'll remove it.

Reply:
Actually, I can't remove them. From my current implementation, I only provide bindings for implemented and existing Reviewable, and Hilt seems to be unaware of any sealed class/closed set of subclasses. With this implementation, I can fix the type I want, and if I don't it won't compile. 
Let's say I want to inject an ItemRepository<T: Reviewable>. Since I cannot provide a generic binding, I can't do it this way (or at least I did not find a simpler way)
I'll let it as this for now.